core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -986,7 +1045,12 @@ private DeploymentPlan makePlanFromYaml(String yaml) {
     }
     
     private CatalogItem<?,?> addItemDto(CatalogItemDtoAbstract<?, ?> itemDto, boolean forceUpdate) {
-        checkItemNotExists(itemDto, forceUpdate);
+        CatalogItem<?, ?> existingDto = checkItemIsDuplicateOrDisallowed(itemDto, true, forceUpdate);
```

Comment:
Name `checkItemIsDuplicateOrDisallowed` is confusing. The `check...` prefix elsewhere follows the guava convention: it throws an exception if the check fails (e.g. `checkNotNull` throws if the condition "not null" fails). Reading this suggests we would throw if "is duplicate or disallowed" fails. But it's the opposite: we're checking that either it's not a duplicate or is allowed.